api/V1/Recording.js

```diff
@@ -0,0 +1,201 @@
+var models = require('../../models');
+var util = require('./util');
+var passport = require('passport');
+var log = require('../../logging');
+var tagsUtil = require('./tagsUtil');
+var requestUtil = require('./requestUtil');
+var responseUtil = require('./responseUtil');
+var multiparty = require('multiparty');
+var config = require('../../config/config')
+var randomstring = require("randomstring");
+var AWS = require('aws-sdk');
+
+module.exports = (app, baseUrl) => {
+  var apiUrl = baseUrl + '/recordings';
+
+  /**
+   * @api {post} /api/v1/recordings Add a new recording.
+   * @apiName PostRecording
+   * @apiGroup Recordings
+   * @apiDescription This call is the new way to upload any type of recording.
+   * It takes a 'data' field which contains a JSON object string that can
+   * contain any of the following fields, Note that 'type' is required:
+   * - type: REQUIRED, one of ('thermalRaw160x120')
+   * - duration
+   * - recordingDateTime
+   * - location
+   * - version
+   * - batteryCharging
+   * - batteryLevel
+   * - airplaneModeOn
+   * - additionalMetadata
+   * - processingMeta
+   *
+   * @apiUse V1DeviceAuthorizationHeader
+   *
+   * @apiParam {JSON} data Metadata about the recording (see above).
+   * @apiParam {File} file File of the recording.
+   *
+   * @apiUse V1ResponseSuccess
+   * @apiuse V1ResponseError
+   */
+  app.post(
+    apiUrl,
+    passport.authenticate(['jwt'], { session: false }),
+    async (request, response) => {
+      log.info(request.method + " Request: " + request.url);
+
+      // Check that the request was authenticated by a device.
+      if (request.user !== null && !requestUtil.isFromADevice(request))
+        return responseUtil.notFromADevice(response);
+      var device = request.user;
+
+      var recording;
+      var key = randomstring.generate();
```

Comment:
A UUID is standard, commonly used way to give a random, unique ID to something. If it's not too much hassle to change the code to use them, it might be nice to switch to them.

Relevant links:
* https://en.wikipedia.org/wiki/Universally_unique_identifier
* https://www.npmjs.com/package/uuid